lib/ExpensiMark.js

```diff
@@ -55,7 +55,13 @@ export default class ExpensiMark {
              */
             {
                 name: 'email',
-                regex: /\[([^[\]]*)\]\(([a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z]+?(\.[a-zA-Z]+)+)\)/gim,
+                process: (textToProcess, replacement) => {
+                    const regex = new RegExp(
+                        '\\[([^[\\]]*)]\\(([a-zA-Z0-9.!#$%&\'*+\\/=?^_`{|}~-]+@[a-zA-Z]+?(\\.[a-zA-Z]+)+)\\)', 'gim'
```

Comment:
This is pretty hard to parse, and when I put it in regex101.com I get a pattern error:
![Screenshot 2022-12-09 at 5 26 29 PM](https://user-images.githubusercontent.com/7277067/206809439-02abb5a9-fc6a-4124-814d-c2d168e768eb.png)

a.) Just curious if you have a different tool you were using to build it/explain it
b.) Maybe we can add some comments or break it up into descriptive variable pieces a bit to make it somewhat more human-readable?

Reply:
Oh, I'm reusing the existing regex, but creating a new Regex required escaping some characters we didn't have to escape before (all those double `\\`). If you make them just one `\`, regex101 will give you all the info:

<img width="684" alt="Screenshot 2022-12-12 at 10 55 15 AM" src="https://user-images.githubusercontent.com/19366280/207015849-372624d2-c406-4d39-92e4-cf420c5432c5.png">

Reply:
Okay I see, thanks!

A couple more thoughts:
1. What do you think about breaking it up for legibility like so:
```js
const firstPartOfEmail = '[a-zA-Z0-9.!#$%&\'*+\\/=?^_`{|}~-]+';
const secondPartOfEmail = '[a-zA-Z]';
const thirdPartOfEmail = '[a-zA-Z]';
const regex = new RegExp(
    `\\[([^[\\]]*)]\\((${firstPartOfEmail}@${secondPartOfEmail}+?(\\.${thirdPartOfEmail}+)+)\\)`, 'gim'
);
```

2. This regex is not allowing for numbers or hyphens in the domain name of the email address, e.g. it doesn't match `alberto@expensify22.com` or `alberto@expens-ify.com`. Perhaps that's outside the scope of this issue though?